src/App.js

```diff
@@ -1,13 +1,51 @@
 import React from 'react';
 
+import { GoodsList } from './components/GoodsList';
+
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+
+export class App extends React.Component {
+  state = {
+    goodsFromServer: [],
+  }
+
+  handleButtonSubmit = (loadGoods) => {
+    loadGoods()
+      .then(goodsFromServer => this.setState({ goodsFromServer }));
+  }
+
+  render() {
+    const { goodsFromServer } = this.state;
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+    return (
+      <>
+        <h1> Dynamic list of Goods</h1>
+        <GoodsList
+          goods={goodsFromServer}
+        />
+        <button
+          type="submit"
+          onClick={() => this.handleButtonSubmit(getAll)}
+        >
+          Load All goods
+        </button>
+        <button
+          type="submit"
+          onClick={() => this.handleButtonSubmit(get5First)}
+        >
+          Load 5 first goods
+        </button>
+        <button
+          type="submit"
```

Comment:
```suggestion
          type="button"
```
Ты же не отправляешь никакую форму никуда, поэтому тут просто button